crates/core/c8y_api/Cargo.toml

```diff
@@ -14,6 +14,7 @@ c8y_smartrest = { path = "../c8y_smartrest" }
 clock = { path = "../../common/clock" }
 csv = "1.1"
 download = { path = "../../common/download" }
+either = "1.6"
```

Comment:
I'm a little bit confused with the use of `either` crate here? Why not go with a straight forward enum of structs?
With the crate's `left` and `right` variants are vague and make reading the code a bit of a hassle to find out which one variant are we using here.

Reply:
Both approaches - using the `either` crate or having some specific implementation, make sense. I agree that what is `Left` or `Right` cannot be guessed without a check to the definition of the `JwtAuthHttpProxy::end_point`. So it might be a good idea switch to a specific implementation - unless one find ourselves re-implementing helper functions . @initard can you check that we mostly use pattern matching to manipulate end-points?

Reply:
I believe it is the case indeed. We should move it to a specific implementation.